src/main/java/com/google/devtools/build/lib/remote/GrpcRemoteExecutor.java

```diff
@@ -129,49 +129,53 @@ public ExecuteResponse executeRemotely(ExecuteRequest request)
     try {
       return retrier.execute(
           () -> {
-            final Iterator<Operation> replies;
-            if (waitExecution.get()) {
-              WaitExecutionRequest wr =
-                  WaitExecutionRequest.newBuilder().setName(operation.get().getName()).build();
-              replies = execBlockingStub().waitExecution(wr);
-            } else {
-              replies = execBlockingStub().execute(request);
-            }
-            try {
-              while (replies.hasNext()) {
-                Operation o = replies.next();
-                operation.set(o);
-                waitExecution.set(!operation.get().getDone());
-                ExecuteResponse r = getOperationResponse(o);
-                if (r != null) {
-                  return r;
-                }
-              }
-            } catch (StatusRuntimeException e) {
-              if (e.getStatus().getCode() == Code.NOT_FOUND) {
-                // Operation was lost on the server. Retry Execute.
-                waitExecution.set(false);
+            for (;;) {
```

Comment:
Why do you feel you need to use an infinite loop as opposed to handling this in the retrier?

Reply:
Because the completion is not an innate exception, and shouldn't count against the retry limit. The existing request behavior - waiting if necessary without a deadline, forever - indicates that a completion doesn't represent an error condition, and that only request exceptions can represent transiency worth retrying and expending retry attempts.

Reply:
Apologies for my comment I was thinking of a different problem that you are trying to fix. Makes total sense. May I ask you to add a test so that we don't regress on it?

Reply:
Test added, fails without change.